File: src/presentation/pages/home_page.js
import { bottomNavShell, mountBottomNav } from '../widgets/bottom_nav_shell.js';

const PROFILE_IMG = 'https://lh3.googleusercontent.com/aida-public/AB6AXuC4YzJzQwQncoyk4L9gJ_ZPPf_AzvNnnkU_o87onjrRsI8ejdwOptD7roe6wRy506s2AhU87sAETjiZw3I7NiLIRH_p9Dj4qHVNQeHXrJkitwcVmPLCBbjt1rfdlvfvC9GW6x4rrS5O-kFzPq3C4tLmwJ9aFkDn67lxlhqRjqw_SnYYYgIylDW5s5tDGI6h0xoJ12dpAqO8wuezx8FO4oFDp_yHIA62C1JNvGJUh8AeJxeFHrqWl5qQjyPfHKpfZGAfWYEblJXITOo';

export function render() {
  return `
    <div class="min-h-screen pb-32">
      <!-- Top App Bar -->
      <header class="bg-surface/80 dark:bg-surface-dim/80 sticky top-0 backdrop-blur-md shadow-sm z-40 transition-all duration-200">
        <div class="flex justify-between items-center px-margin-mobile h-16 w-full">
          <div class="flex items-center gap-md">
            <img src="${PROFILE_IMG}" alt="User Profile" class="w-10 h-10 rounded-full object-cover border-2 border-primary-container">
            <h1 class="font-headline-sm text-headline-sm text-primary">Gait Analysis</h1>
          </div>
          <button class="text-primary hover:opacity-80 active:scale-95 transition-all">
            <span class="material-symbols-outlined">sensors</span>
          </button>
        </div>
      </header>

      <main class="px-margin-mobile pt-lg space-y-xl">
        <!-- Greeting -->
        <section class="space-y-xs">
          <p class="font-body-md text-on-surface-variant">Hello, welcome back.</p>
          <h2 class="font-headline-md text-headline-md text-on-surface">Ready for your gait assessment?</h2>
        </section>

        <!-- Hero Card -->
        <section class="relative overflow-hidden bg-teal-gradient rounded-lg shadow-lg p-lg text-white">
          <div class="relative z-10 w-2/3 space-y-md">
            <h3 class="font-headline-sm text-headline-sm">Start a new gait test</h3>
            <p class="font-body-md opacity-90">Record your walking data using your phone sensors.</p>
            <button id="start-test-btn" class="bg-on-surface text-surface py-md px-lg rounded-full font-label-md hover:opacity-90 active:scale-95 transition-all">
              Start Test
            </button>
          </div>
          <div class="absolute top-0 right-0 h-full w-1/3 flex items-center justify-center opacity-40">
            <span class="material-symbols-outlined !text-[120px]" style="font-variation-settings: 'wght' 200;">directions_walk</span>
          </div>
        </section>

        <!-- Stats Bento Grid -->
        <section class="grid grid-cols-2 gap-md">
          <!-- FMA-LE Score -->
          <div class="col-span-2 bg-surface-container-lowest p-lg rounded-lg shadow-[0_4px_20px_0_rgba(29,53,87,0.06)] border border-outline-variant/20">
            <div class="flex justify-between items-start mb-md">
              <span class="font-label-sm text-on-surface-variant uppercase tracking-wider">Latest Score</span>
              <span class="px-sm py-1 bg-primary-container/20 text-on-primary-container text-xs rounded-full font-bold">Moderate</span>
            </div>
            <div class="flex items-end gap-base">
              <span class="font-display-lg text-display-lg text-primary">24</span>
              <span class="font-body-md text-on-surface-variant mb-1">FMA-LE</span>
            </div>
            <p class="mt-md font-body-md text-on-surface-variant text-sm">Improvement of +2 pts from last week.</p>
          </div>

          <!-- Classification -->
          <div class="bg-surface-container-lowest p-md rounded-lg shadow-[0_4px_20px_0_rgba(29,53,87,0.06)] border border-outline-variant/20 flex flex-col justify-between aspect-square">
            <div>
              <span class="material-symbols-outlined text-secondary" style="font-variation-settings: 'FILL' 1;">analytics</span>
              <h4 class="font-label-sm text-on-surface-variant mt-sm">Last Classification</h4>
            </div>
            <div>
              <p class="font-headline-sm text-on-surface">CVA</p>
              <p class="font-label-sm text-primary">91% confidence</p>
            </div>
          </div>

          <!-- Sessions -->
          <div class="bg-surface-container-lowest p-md rounded-lg shadow-[0_4px_20px_0_rgba(29,53,87,0.06)] border border-outline-variant/20 flex flex-col justify-between aspect-square">
            <div>
              <span class="material-symbols-outlined text-secondary" style="font-variation-settings: 'FILL' 1;">calendar_today</span>
              <h4 class="font-label-sm text-on-surface-variant mt-sm">Total Sessions</h4>
            </div>
            <div class="flex items-baseline gap-xs">
              <p class="font-display-lg text-display-lg text-on-surface">12</p>
              <p class="font-label-sm text-on-surface-variant">tests</p>
            </div>
          </div>
        </section>

        <!-- Recent Activity -->
        <section class="flex justify-between items-center">
          <h3 class="font-headline-sm text-headline-sm text-on-surface">Activity History</h3>
          <button data-nav-to="/history" class="text-primary font-label-md">View All</button>
        </section>

        <div class="space-y-md">
          <div class="bg-surface-container-lowest p-md rounded-lg flex items-center gap-md border border-outline-variant/10">
            <div class="w-12 h-12 bg-surface-container-highest rounded-2xl flex items-center justify-center text-primary">
              <span class="material-symbols-outlined">directions_walk</span>
            </div>
            <div class="flex-1">
              <p class="font-label-md text-on-surface">Gait Test #012</p>
              <p class="font-label-sm text-on-surface-variant">Yesterday, 4:30 PM</p>
            </div>
            <div class="text-right">
              <p class="font-data-viz text-primary">88%</p>
              <p class="font-label-sm text-on-surface-variant text-[10px]">SYMMETRY</p>
            </div>
          </div>
        </div>
      </main>

      ${bottomNavShell('home')}
    </div>
  `;
}

export function mount(navigate) {
  mountBottomNav(navigate);
  document.getElementById('start-test-btn')?.addEventListener('click', () => navigate('/instruction'));
}

File: lib/presentation/pages/home_page.dart
import 'package:flutter/material.dart';
import 'package:go_router/go_router.dart';
import 'package:provider/provider.dart';
import '../../core/constants/app_colors.dart';
import '../../core/constants/app_routes.dart';
import '../../core/constants/app_text_styles.dart';
import '../../core/theme/app_theme.dart';
import '../providers/session_provider.dart';
import '../providers/profile_provider.dart';
import '../widgets/app_card.dart';
import '../widgets/bottom_nav_shell.dart';
import '../widgets/custom_app_bar.dart';

class HomePage extends StatelessWidget {
  const HomePage({super.key});

  @override
  Widget build(BuildContext context) {
    final sessions = context.watch<SessionProvider>().history;
    final profile = context.watch<ProfileProvider>().profile;
    final displayName =
        profile.name.trim().isEmpty ? 'there' : profile.name.trim();

    return Scaffold(
      extendBodyBehindAppBar: true,
      appBar: CustomAppBar(
        title: 'Gait Analysis',
        actions: [
          IconButton(
            icon: const Icon(Icons.sensors, color: AppColors.primary),
            onPressed: () {},
          ),
        ],
      ),
      body: CustomScrollView(
        slivers: [
          SliverToBoxAdapter(
            child: SizedBox(
                height: 64 + MediaQuery.of(context).padding.top + 8),
          ),
          SliverPadding(
            padding: const EdgeInsets.symmetric(
                horizontal: AppTheme.marginMobile),
            sliver: SliverList(
              delegate: SliverChildListDelegate([
                // Greeting
                Text('Hello, welcome back',
                    style: AppText.bodySm),
                const SizedBox(height: 4),
                Text(displayName, style: AppText.headlineSm),
                const SizedBox(height: AppTheme.lg),
                // Hero card
                _HeroCard(onStartTest: () => context.go(AppRoutes.instruction)),
                const SizedBox(height: AppTheme.lg),
                // Bento stats grid
                _StatsGrid(),
                const SizedBox(height: AppTheme.xl),
                // Activity history header
                Row(
                  mainAxisAlignment: MainAxisAlignment.spaceBetween,
                  children: [
                    Text('Recent Activity', style: AppText.headlineSm),
                    TextButton(
                      onPressed: () => context.go(AppRoutes.history),
                      child: Text('View All',
                          style: AppText.labelMd
                              .copyWith(color: AppColors.primary)),
                    ),
                  ],
                ),
                const SizedBox(height: AppTheme.md),
                // Session cards
                ...sessions.take(3).map((s) => Padding(
                      padding: const EdgeInsets.only(bottom: AppTheme.sm),
                      child: _SessionCard(
                        date: _formatDate(s.date),
                        classification: s.classification,
                        severity: s.severity,
                        fmaScore: s.result.fmaLeScore,
                        symmetry: (s.result.semiogram.symmetryIndex * 100)
                            .round(),
                        onTap: () => context.go(AppRoutes.result),
                      ),
                    )),
                const SizedBox(height: 100),
              ]),
            ),
          ),
        ],
      ),
      bottomNavigationBar: const BottomNavBar(activePage: 'home'),
    );
  }

  String _formatDate(DateTime d) =>
      '${d.day.toString().padLeft(2, '0')} ${_monthName(d.month)} ${d.year}';

  String _monthName(int m) => const [
        '', 'Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun',
        'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'
      ][m];
}

class _HeroCard extends StatelessWidget {
  final VoidCallback onStartTest;
  const _HeroCard({required this.onStartTest});

  @override
  Widget build(BuildContext context) {
    return Container(
      height: 180,
      decoration: BoxDecoration(
        gradient: const LinearGradient(
          begin: Alignment.topLeft,
          end: Alignment.bottomRight,
          colors: [Color(0xFF006A62), Color(0xFF2EC4B6)],
        ),
        borderRadius: AppTheme.radiusLg,
        boxShadow: AppTheme.primaryShadow,
      ),
      clipBehavior: Clip.hardEdge,
      child: Stack(
        children: [
          Positioned(
            right: -20,
            bottom: -20,
            child: Icon(
              Icons.directions_walk,
              size: 160,
              color: Colors.white.withAlpha(26),
            ),
          ),
          Padding(
            padding: const EdgeInsets.all(AppTheme.lg),
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              mainAxisAlignment: MainAxisAlignment.center,
              children: [
                Text('Start a new gait test',
                    style: AppText.headlineSm
                        .copyWith(color: Colors.white)),
                const SizedBox(height: 4),
                Text('Attach sensor and begin walking',
                    style: AppText.bodySm
                        .copyWith(color: Colors.white70)),
                const SizedBox(height: AppTheme.md),
                GestureDetector(
                  onTap: onStartTest,
                  child: Container(
                    padding: const EdgeInsets.symmetric(
                        horizontal: 20, vertical: 10),
                    decoration: BoxDecoration(
                      color: Colors.white,
                      borderRadius: AppTheme.radiusFull,
                    ),
                    child: Text(
                      'Start Test',
                      style: AppText.labelMd
                          .copyWith(color: AppColors.primary),
                    ),
                  ),
                ),
              ],
            ),
          ),
        ],
      ),
    );
  }
}

class _StatsGrid extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    final history = context.watch<SessionProvider>().history;
    final latest = history.isNotEmpty ? history.first : null;

    return Column(
      children: [
        // FMA-LE card (full width)
        AppCard(
          child: Row(
            children: [
              Expanded(
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    Text('FMA-LE Score',
                        style: AppText.labelSm),
                    const SizedBox(height: 4),
                    Text(
                        latest != null
                            ? latest.result.fmaLeScore.toString()
                            : '—',
                        style: AppText.dataLg
                            .copyWith(color: AppColors.primary)),
                    Text('out of 34',
                        style: AppText.labelSm),
                  ],
                ),
              ),
              if (latest != null)
                _SeverityBadge(severity: latest.severity),
            ],
          ),
        ),
        const SizedBox(height: AppTheme.sm),
        Row(
          children: [
            // Classification
            Expanded(
              child: AppCard(
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    Text('Classification',
                        style: AppText.labelSm),
                    const SizedBox(height: 4),
                    Text(latest != null ? latest.classification : '—',
                        style: AppText.headlineSm
                            .copyWith(color: AppColors.primary)),
                    const SizedBox(height: 4),
                    Text(
                        latest != null
                            ? '${(latest.result.confidence * 100).round()}% confidence'
                            : 'No tests yet',
                        style: AppText.labelSm
                            .copyWith(color: AppColors.success)),
                  ],
                ),
              ),
            ),
            const SizedBox(width: AppTheme.sm),
            // Sessions
            Expanded(
              child: AppCard(
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    Text('Sessions', style: AppText.labelSm),
                    const SizedBox(height: 4),
                    Text(history.length.toString(),
                        style: AppText.headlineSm
                            .copyWith(color: AppColors.secondary)),
                    const SizedBox(height: 4),
                    Text('total tests',
                        style: AppText.labelSm),
                  ],
                ),
              ),
            ),
          ],
        ),
      ],
    );
  }
}

class _SeverityBadge extends StatelessWidget {
  final String severity;
  const _SeverityBadge({required this.severity});

  Color get _color => switch (severity.toLowerCase()) {
        'mild' => AppColors.success,
        'moderate' => AppColors.warning,
        'severe' => AppColors.error,
        _ => AppColors.onSurfaceVariant,
      };

  Color get _bg => switch (severity.toLowerCase()) {
        'mild' => AppColors.successContainer,
        'moderate' => AppColors.warningContainer,
        'severe' => AppColors.errorContainer,
        _ => AppColors.surfaceContainerHighest,
      };

  @override
  Widget build(BuildContext context) {
    return Container(
      padding: const EdgeInsets.symmetric(horizontal: 12, vertical: 6),
      decoration: BoxDecoration(
        color: _bg,
        borderRadius: AppTheme.radiusFull,
      ),
      child: Text(severity,
          style: AppText.labelSm.copyWith(
              color: _color, fontWeight: FontWeight.w700)),
    );
  }
}

class _SessionCard extends StatelessWidget {
  final String date;
  final String classification;
  final String severity;
  final int fmaScore;
  final int symmetry;
  final VoidCallback onTap;

  const _SessionCard({
    required this.date,
    required this.classification,
    required this.severity,
    required this.fmaScore,
    required this.symmetry,
    required this.onTap,
  });

  @override
  Widget build(BuildContext context) {
    return AppCard(
      onTap: onTap,
      child: Column(
        children: [
          Row(
            mainAxisAlignment: MainAxisAlignment.spaceBetween,
            children: [
              Expanded(
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    Text(date, style: AppText.labelSm),
                    const SizedBox(height: 2),
                    Text(classification, style: AppText.labelMd),
                  ],
                ),
              ),
              _SeverityBadge(severity: severity),
            ],
          ),
          const SizedBox(height: AppTheme.sm),
          Row(
            children: [
              Expanded(
                child: _MetricItem(
                    label: 'FMA-LE',
                    value: '$fmaScore / 34'),
              ),
              Expanded(
                child: _MetricItem(
                    label: 'Symmetry',
                    value: '$symmetry%'),
              ),
            ],
          ),
        ],
      ),
    );
  }
}

class _MetricItem extends StatelessWidget {
  final String label;
  final String value;
  const _MetricItem({required this.label, required this.value});

  @override
  Widget build(BuildContext context) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        Text(label, style: AppText.labelSm),
        Text(value,
            style: AppText.dataViz.copyWith(color: AppColors.primary)),
      ],
    );
  }
}
